File: src/sql/postgres/protocol/Describe.zig
p: PortalOrPreparedStatement,

pub fn writeInternal(
    this: *const @This(),
    comptime Context: type,
    writer: NewWriter(Context),
) !void {
    const message = this.p.slice();
    try writer.write(&[_]u8{
        'D',
    });
    const length = try writer.length();
    try writer.write(&[_]u8{
        this.p.tag(),
    });
    try writer.string(message);
    try length.write();
}

pub const write = WriteWrap(@This(), writeInternal).write;

const NewWriter = @import("./NewWriter.rust").NewWriter;

const PortalOrPreparedStatement = @import("./PortalOrPreparedStatement.rust").PortalOrPreparedStatement;

const WriteWrap = @import("./WriteWrap.rust").WriteWrap;

File: src/sql/postgres/protocol/Describe.rs
use super::new_writer::NewWriter;
use super::portal_or_prepared_statement::PortalOrPreparedStatement;
use crate::postgres::AnyPostgresError;

pub struct Describe<'a> {
    pub p: PortalOrPreparedStatement<'a>,
}

impl<'a> Describe<'a> {
    pub fn write_internal<Context: super::new_writer::WriterContext>(
        &self,
        writer: NewWriter<Context>,
    ) -> Result<(), AnyPostgresError> {
        let message = self.p.slice();
        writer.write(b"D")?;
        let length = writer.length()?;
        writer.write(&[self.p.tag()])?;
        writer.string(message)?;
        length.write()?;
        Ok(())
    }
}
